### PR TITLE
Make JobStore impls use the new PortabilityJob for #126

### PR DESCRIPTION
**Pull request by rtannenbaum**

### CODE REVIEW

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java [hunk 1]
```diff
@@ -88,29 +86,62 @@ public void create(UUID jobId, LegacyPortabilityJob job) throws IOException {
   }
 
   /**
-   * Finds the {@link LegacyPortabilityJob} keyed by {@code jobId} in Datastore, or null if none found.
+   * Atomically updates the {@link LegacyPortabilityJob} keyed by {@code jobId} to {@code job},
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java]: s/LegacyPortabilityJob/PortabilityJob

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java [hunk 2]
```diff
@@ -88,29 +86,62 @@ public void create(UUID jobId, LegacyPortabilityJob job) throws IOException {
...
+  public void updateJob(UUID jobId, PortabilityJob job) throws IOException {
+    Preconditions.checkNotNull(jobId);
+    Transaction transaction = datastore.newTransaction();
+    Key key = getKey(jobId);
+
+    try {
+      Entity previousEntity = transaction.get(key);
+      if (previousEntity == null) {
+        transaction.rollback();
+        throw new IOException("Could not find record for jobId " + jobId);
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java]: let's do jobId: " + jobId to distinguish parameterized values or any convention to highlight for future debugging

Comment (holachuy) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java]: Here and below.

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java (hunk 1 above)
Comment (rtannenbaum) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java]: Done

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java (hunk 2 above)
Comment (rtannenbaum) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudDatastore.java]: Good point, done

**Pull request merged**